daml-lf/spec/daml-lf-1.rst

```diff
@@ -3554,6 +3554,15 @@ The following builtin functions defines an order on the so-called
 abstractions, functions, partially applied builtin functions, and
 updates.
 
+Note that as described in the `V1 Contract ID allocation scheme
+specification <./contract-id.rst>`_ the comparison of two V1 contract
+identifiers may fail at run time. For the purpose of this
+specification, we will say that two contract identifiers are *not
+comparable* if (i) both of them are V1 contract identifiers, (ii) one
```

Comment:
I’m a bit confused by the condition here, you have i && ii && iii but I would have expected !i || (ii && iii) meaning either one or both is a v0 cid or both are v1 cids and 3 apply.

Reply:
I am problably way more confused that you ;) .

Note that here I am specifying *not comparable* (maybe it is a bad idea).


Let me redefine the opposite condition:
 
Two cids are comparable if one of the following condition is true: 
- at least one of them is V0: `!(i)`
- both are suffixed : `!(ii)` 
- none of them is a prefix of the other : `!(iii)`

So we have cids are comparable if `!(i)` || `!(ii)` || `!(iii)` or cids are not comparable if  `(i) && (ii) && (iii)`.

Reply:
Does this make sens.

Reply:
Doesn’t make sense to me. Why are v0 cids comparable? I thought comparisons require v1?
or is comparable only equality but not ordering?

Reply:
Recording our conversation: It does make sense, we do allow for comparisons of v0 contract ids since they are always global (or you run in an engine that does not provide comparison primitives).